src/Parser/BalanceDruid/Modules/Features/AstralPower.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import Icon from 'common/Icon';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Module from 'Parser/Core/Module';
+import SPELLS from 'common/SPELLS';
+
+class AstralPower extends Module {
+  
+  lastAstral = 0;
+  aspWasted = 0;
+
+  on_toPlayer_energize(event) {
+      for (let i = 0; i < event.classResources.length; i++) {
+          if (event.classResources[i].type === 8){
+              const maxAsP = event.classResources[i].max / 10;
+              const addedAsP = event.resourceChange;
+
+              if (this.lastAstral + addedAsP > maxAsP){
+                  this.aspWasted += this.lastAstral + addedAsP - maxAsP;
+              }
+
+              this.lastAstral = event.classResources[i].amount / 10;
+          }
+      }
+  }
+
+  on_byPlayer_cast(event) {
+      const spellId = event.ability.guid;
+      if (SPELLS.STARSURGE_MOONKIN.id !== spellId && SPELLS.STARFALL.id !== spellId && SPELLS.STARFALL_CAST.id !== spellId) {
+          return;
+      }
+      
+      for (let i = 0; i<event.classResources.length; i++) {
+          if (event.classResources[i].type === 8){
+              if (event.classResources[i].cost) {
+                  this.lastAstral = this.lastAstral - (event.classResources[i].cost / 10);
+              }
+          }
+      }
+  }
+
+  suggestions(when) {
+      when(this.aspWasted).isGreaterThan(0)
+        .addSuggestion((suggest, actual, recommended) => {
+          return suggest(<span>You overcapped {this.aspWasted} Astral Power.</span>)
+            .icon('ability_druid_cresentburn')
+            .actual(`${this.aspWasted} overcapped Astral Power`)
+            .recommended(`You shouldn't overcap any, always prioritize spending it over avoiding the overcap or any other ability.`)
```

Comment:
`<${recommended} is recommended` would be much more concise. The main text is for giving advice, this is mostly just the technical way to inform users.

Reply:
As said before, there's no a recomended range of AsP wasted, you should plain not waste any.
Too many people ask about the priority between overcapping other spells or AsP, which is the main reason for this wording, just trying to be as useful as possible for people trying to find our what's wrong with their rotation.
Will change it if it's a big problem tho.

Reply:
Makes sense, part of the reason I'd like to see this shortened is since I also recommended moving the suggestion part to the suggestion text (https://github.com/MartijnHols/WoWAnalyzer/pull/226/files/ba4c6490d3f79129c1143dc8478fddc20fbdd0c9#diff-919504bd102993275ef3c775a1bfd8e1R45), so I think this should be as concise as possible. How about `0 overcapped Astral Power is recommended`?